### ISSUE TITLE
Feature Request: Dynamic parameter handling for directories

### ISSUE THREAD

**Issue by jgable** [krakenjs/express-enrouten#35]
I'm often in situations where I'd like to be able to separate out some deeply nested routes based on a `root/:param` type of situation.

Here are some example routes to illustrate the points:
- `/api/projects`
- `/api/projects/:taskId`
- `/api/projects/:taskId/team`
- `/api/projects/:taskId/team/:teamId`

Right now, I'd have one big `index.js` file in the `/api/projects` directory for all those routes.  What do you think about a structure like this:

``` shell
/api
  /projects
    index.js
    /_taskId
      index.js
      /team
        index.js
        /_teamId
          index.js
```

Basically, directories with a leading _ would be converted to parameter routes.  I mean, I'll admit it gets a little deep directory wise.  But, you could always still just put everything right in the index.js.  I think coupled with some CLI tooling/generators this wouldn't be that bad to whip out deep API handlers pretty quickly.

**Comment by blowsie** [comment]
late to the party here, but Nuxt has this style of routing
